wai-app-static/Network/Wai/Application/Static.hs

```diff
@@ -214,10 +214,10 @@ staticApp set req = staticAppPieces set (W.pathInfo req) req
 
 staticAppPieces :: StaticSettings -> [Text] -> W.Application
 staticAppPieces _ _ req sendResponse
-    | W.requestMethod req /= "GET" = sendResponse $ W.responseLBS
+    | notElem (W.requestMethod req) ["GET", "HEAD"] = sendResponse $ W.responseLBS
```

Comment:
I didn't notice the branch, thanks for pointing that out. OK, this is good to merge, thank you!